### ISSUE TITLE
tamil_lexicon discussion 

### ISSUE THREAD

**Issue by damooo** [indic-dict/stardict-tamil#2]
Tamil_lexicon added.

**Comment by damooo** [comment]
i am confused with tamil alphabet transliteration as there are many special characters, and same letter for three or four sounds.. and online transliterators provide each differant thing, and some online transliterators like sanscript not giving transliteration for some chars at all . but dictionary itself have pronounciational transliteration. i added that one too in headers. though special chars used in them, goldendict providing results for normal optitrans too. so along with them if i got a chart with reliable transliterations, i will add optitrans too.

**Comment by vvasuki** [comment]
@damooo - 
Nice work! But please consider the below:

=============
# Style bloat
I suspect that if you remove unnecessary css and html you won't be hitting the 100MB limit at all. 

Please compare:
```
அ_2|அ|a
<style>span.shirshika{font-weight:bold;font-size:1.5em;} span.uchcharana{color:blue;} span.vyutpatti{color:red;} a.tl_pada:link,a.tl_pada:visited,a.tl_pada{ text-decoration:none; color:#2f4f4f; }a.tl_pada:hover{ text-decoration: underline; color:#2f4f4f; } div.tl_patra{margin: 0 auto 0 auto; line-height:1.5em;padding: 1em; max-width:55.8em; font-size: 1em; padding:0.1em;} sup{font-size:0.5em;}</style><div class="tl_patra"><span class="shirshika">அ<sup>2</sup></span><hr><span class="uchcharana">a</span><br><span class="vyutpatti">n.</span><br>Symbol for the number 8, printed without the loop at the top (<a class="tl_pada" href="அ">அ</a>);<br>எட்டென்னும் எண்ணின் <a class="tl_pada" href="குறி">குறி</a>. <a class="tl_pada" href="அ">அ</a> உ அறியா(யாப்.<a class="tl_pada" href="வி">வி</a>.37, <a class="tl_pada" href="உரை">உரை</a>)<p></p></div>

```
versus
`அ_2   அ   a<br>==================<br>Symbol for the number 8, printed without the loop at the top (<a href="அ">அ</a>);<br>எட்டென்னும் எண்ணின் <a href="குறி">குறி</a>. <a href="அ">அ</a> உ அறியா(யாப்.<a href="வி">வி</a>.37, <a href="உரை">உரை</a>)<p></p></div>
`
versus (the anchors are quite useless since you can long press or double click on **any** text and the dictionary will easily take you there.)
`அ_2|அ|a<br>==================<br>Symbol for the number 8, printed without the loop at the top (அ);<br>எட்டென்னும் எண்ணின் குறி. அ உ அறியா(யாப். வி .37, உரை)<p></p></div>
`
Now consider if it is a good idea to strain the memory of users' phones (and face trouble with github space) for styling that doesn't really make such a big difference when it comes to looking up words. In the meantime, there is possibility that users still cannot get basic information from a dictionary as happens [here](https://github.com/sanskrit-coders/stardict-sanskrit/issues/23).

Also look at this issue page - what big stylesheet are we using when we type these markdown syntax comments. This makes me wonder - **can we just use markdown instead of css**?
=============================
Please also use devanAgarI headwords.
=============================
# About the script:
The alphabet is very simple - 
for vargIya vyanjana-s, the same symbol is used for the first four letters. For example, क ख ग घ ङ = க க க க ங . Modern writers use subscripts to differentiates.
Other than that there is La, which telugu also has, and in addition LLa, RRa and NNa - for these, it is fine to use the symbols LL RR and NN while doing optitrans. Also, this may be useful - http://unicode.org/charts/PDF/U0B80.pdf ,

**Comment by vvasuki** [comment]
Also, please look up <https://github.com/git-lfs/git-lfs/tree/master/docs> (I haven't been able to get it working yet.)

**Comment by vvasuki** [comment]
Please also note the breakage in https://github.com/sanskrit-coders/stardict-telugu/issues/13#issuecomment-289644273

**Comment by damooo** [comment]
Atleast anchors are necessary, as they are to the words in that dictionary itself. and quite helpful on many practical scenarios, and other uses;
I tried to be too minimal, with just highlighting very practical,useful differances.. just four instructions used. many friends felt that minimal classificational differentiation is needed for practical usage.. and hense kept those four instructions.. 
or can we provide two versions, one with style(will keep it to just basic needs) for desktop users, and other people who ever want that; and other non html text version for others..

**Comment by vvasuki** [comment]
> Atleast anchors are necessary, as they are to the words in that dictionary itself. and quite helpful on many practical scenarios, and other uses;

Anchors, though unnecessary in case of words linked without any change for users who know about long-press, can be retained - no problems with that.

> I tried to be too minimal, with just highlighting very practical,useful differances.. just four instructions used. 
I appreciate that.. but your idea of minimal seems very very different from mine.

Let me show you what I mean:
![image](https://cloud.githubusercontent.com/assets/2664797/24389560/5f2ddf88-1337-11e7-925b-74a4d9965eb0.png)
Both those dictionaries were made by you. The bottom one *has* style (text based, not css), it is clear. It is relatively small, and gets the job done well enough. Users come to look at dictionary entries, not to appreciate stylized headwords and the like. The top one adds no value.

> many friends felt that minimal classificational differentiation is needed for practical usage.. and hense kept those four instructions..
Please consider whether your many friends are just telling you what makes you happy, or if they know the downsides of having this additional styles. They may not be representative sample of the actual user population. I'd rather go with something which is guaranteed to get the simple word-lookup job done, without these extra space problems ---- until users complain about lack of style. This style thing breaks whenever (and remains broken until we find time to fix it), that is a far bigger risk.

If despite this, our convictions about this differ, we can do this: just make a no-style dict ( - but plz retain the links) which I can distribute. The other high style version - we can just keep in a subdirectory - it will be useful for desktop users or anyone else who prefers that on other devices.

**Comment by damooo** [comment]
pushed stripped versions.. plz rearrange files,folders in desired way. 

i don't use styles to look better, or feel happy, but to highlight differant sections. i can use bare text for paribhashika too, but that is not going to work well. reasons are:
1. every word in results is to be hyperlinked as they are individual words rather than description. And then if not styled, everything is blue, underlined, unusable.  otherwise suggest a solution.
2. and then there are sections for many words like one section for namavachaka, other for vishEShaNa usage etc, which need to be proved in appearance. otherwise everything mixup.
3. and in each of above section, each horizental row represents a differant  set of meanings, which again have to be properly organised, irrespective of screensize, and have to be sensed by seeing. hense that line grouping too done.
your search word only has one line, no nanartha set, no bhashabhaga grouping. most of words have them. search like 'address' , etc.

there is no much use of this dictionary with clumsy mix, loosing all relationship between words, and grouping among them.
in other words i use html only to prove/preserve the intended grouping, sectioning of book. otherwise i don't do.
to mass create above structure from pdf, i have to work hard, i didn't did it for feeling happy, but for representing,preserving it's intentional use. like footnotes in veda index, para deapths,side headings, header sectioning  in pe, sections in paribhashika. page numbers in historical dicts. hyperlinks in all. etc. They can be used in seconds if any one host them in website. with representing all info. otherwise, they all looks like juggled , unfinished text, which cannot be accessed by machines,softwears, and not usable further.

otherwise tags, positioning, styling is least concern for me.

**Comment by vvasuki** [comment]
Thank you, @damooo .  I appreciate that your very significant efforts are also done with the user in mind - I did not say that you were doing that to feel good (which is also an important goal in anything). I was just saying that the feedback you received from your friends could be biased, that's all. Please don't take it otherwise. 

> to mass create above structure from pdf, i have to work hard, i didn't did it for feeling happy, but for representing,preserving it's intentional use. 

I appreciate your hard work, and I know the feeling from experience in similar projects 🙏 

> like footnotes in veda index, para deapths,side headings, header sectioning in pe, sections in paribhashika. page numbers in historical dicts. hyperlinks in all. etc. 

The pagenumbers were just brilliant, and hyperlinks are useful in many cases. The handling of footnotes make me a big fan of your work - now that is just a splendid example of absolutely necessary styling.
About the sections in pAribhAShika, please see below.


# Trying to understand the style requirements a bit better.
> every word in results is to be hyperlinked as they are individual words rather than description. And then if not styled, everything is blue, underlined, unusable. otherwise suggest a solution.

Can you show me an example entry? One that I looked at seemed fine without the style - 
![image](https://cloud.githubusercontent.com/assets/2664797/24411888/dbdcda3a-138b-11e7-9f19-257cc0362d4a.png)

If it turns out that *every* word is to be hyperlinked, we are doing something wrong - the dictionary viewer should and does make it convenient to select and look up words.

> and then there are sections for many words like one section for namavachaka, other for vishEShaNa usage etc, which need to be proved in appearance. otherwise everything mixup.

An example here too please..

> and in each of above section, each horizental row represents a differant set of meanings, which again have to be properly organised, irrespective of screensize, and have to be sensed by seeing. hense that line grouping too done.

I think that you are trying to do the dictionary viewer's job. The dict viewer should take care of rendering the dicts as the user wants, not so much the dictionary.

> your search word only has one line, no nanartha set, no bhashabhaga grouping. most of words have them. search like 'address' , etc.
You mean like this?
![image](https://cloud.githubusercontent.com/assets/2664797/24412132/7585cf34-138c-11e7-9593-3e7ab0c02de7.png)
![image](https://cloud.githubusercontent.com/assets/2664797/24412664/f27f80a6-138d-11e7-9f54-0be453a76c8c.png)

That text-only entry is not horrible, and can be made better in several ways:
* Make a separate entry for each artha. <--- best way.
* use simple html (lists and horizontal rules) without the risky css bloat.
* add better text styles and separators

# Other requests

I think you missed other requests I made in a previous comment - particularly adding devanAgarI headwords. In any case, I will make sure to fix my scripts and distribute your work as soon as possible.

Btw, how did you manage to get the babylon size below 100MB for the styled version now?

**Comment by damooo** [comment]
i learned tamil script recently so that i can read and write, and do work with it. soon will add improvements and other dicts.
